juju/juju
# [JUJU-4305] Inject ctrlConfigService in client/controller facade

This PR replaces all legacy (state-based) ControllerConfig API calls with the ControllerConfig Domain calls in client/controller facade. Need to mention that this is the only place in the Juju codebase,
where we need to write Controller Config to DB.


*Why this change is needed and what it does.*

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

*Commands to run to verify that the change works.*

Unit-tests should pass:
```sh
go test github.com/juju/juju/apiserver/facades/client/controller/... -gocheck.v
```

Controller config should be in Dqlite DB:
```sh
juju bootstrap lxd lxd
juju switch controller
juju ssh 0
# in machine 0
apt install go-dqlite
dqlite -s 127.0.0.1:17666 controller
select * from controller_config
# Observ controller config data

```
